// containers/TableContainer------workingCOPY.js
import React, { Component } from 'react';
import Search from 'components/Search';
import Table from 'components/Table';
import localData from 'assets/local-data';
import {parseResponseInNewArray, patternSorting} from 'myHelpers';
import myHistory from 'myHistory';

// import userRequestActions from 'actions/userRequestActions';

export default class extends Component {
  state = {
    dataParsedInArr: localData,
    nameSortTest: false,
    ratingSortTest: false
  };


  componentWillMount() {
    // лолучение данных тут
    //----------------------------------------------------

    const newArr = parseResponseInNewArray(localData);
    this.setState({dataParsedInArr: newArr});
  }

  componentDidMount() {
    // подписка на события хранилища
    // и возможно лолучение данных тут
    //----------------------------------------------------

  }

  componentWillUnmount() {
    // Отписка от событий хранилища
    //----------------------------------------------------

  }

  changeRequest = (cb) => {
    console.log('передаю назад', cb );
  };
// -------------------------------- END ---------------------------------


  openDetails = (event) => {
    if (event.target.parentElement.classList.contains('table__row-data')) {
      const id = +event.target.parentElement.dataset.id;
      return myHistory.push(`/details/:id${id}`)
    }
  };
  // -------------------------------- END ---------------------------------


  handleSortName = () => {
    this.setState({
      nameSortTest: !this.state.nameSortTest,
      dataParsedInArr: patternSorting(
        this.state.dataParsedInArr,
        this.state.nameSortTest,
        'str',
        1
      )
    })
  };


  handleSortRating = () => {
    this.setState({
      ratingSortTest: !this.state.ratingSortTest,
      dataParseInArrow: patternSorting(
        this.state.dataParsedInArr,
        this.state.ratingSortTest,
        'num',
        5
      )
    })
  };


  render() {
    return (
      <div>
        <button onClick={this.handleSortName}>toggle str</button>
        <button onClick={this.handleSortRating}>toggle num</button>
        <Search onChangeRequest={this.changeRequest} />
        <Table ocClickTable={this.openDetails} data={this.state.dataParsedInArr}/>
      </div>
    );
  }
}
