// containers/TableContainer.js
import React, { Component } from 'react';
import Search from 'components/Search';
import Table from 'components/Table';
import localData from 'assets/local-data';
import {parseResponseInNewArray, patternSorting} from 'myHelpers';

// import userRequestActions from 'actions/userRequestActions';

export default class extends Component {
  state = {
    dataParsedInArr: localData,
    nameSortTest: false,
    ratingSortTest: false
  };


  componentWillMount() {
    const newArr = parseResponseInNewArray(localData);
    this.setState({dataParsedInArr: newArr});
  }



  changeRequest = (cb) => {
    console.log('передаю назад', cb );
  };
// -------------------------------- END ---------------------------------


  handleSortName = () => {
    this.setState({
      nameSortTest: !this.state.nameSortTest,
      dataParsedInArr: patternSorting(
        this.state.dataParsedInArr,
        this.state.nameSortTest,
        'str',
        0
      )
    })
  };


  handleSortRating = () => {
    this.setState({
      ratingSortTest: !this.state.ratingSortTest,
      dataParseInArrow: patternSorting(
        this.state.dataParsedInArr,
        this.state.ratingSortTest,
        'num',
        4
      )
    })
  };

  openDetails = (event) => {
    console.log('принял евент с таблицы====', event);
  };

  render() {
    return (
      <div>
        <button onClick={this.handleSortName}>toggle str</button>
        <button onClick={this.handleSortRating}>toggle num</button>
        <Search onChangeRequest={this.changeRequest} />
        <Table ocClickTable={this.openDetails} data={this.state.dataParsedInArr}/>
      </div>
    );
  }
}
