// pages/Details/index.js
import React, {Component} from 'react';
import './index.scss'

// export default (props) => {
export default class extends Component {
  state = {
    responseData: this.props.state.data.responseData,
    data: null
  };

  componentWillMount() {
    const urlID2 = window.location.href.match(/(id\d+)/)[0].split('').splice(2).join('');

    if (!!this.props.state.data.responseData) {
      this.setState({
        data: JSON.parse(localStorage.getItem('data-details'))
      });
    }
    for (let i = 0; i < Object.keys(this.state.responseData).length; i++) {
      if (this.state.responseData[i].show.id === +urlID2) {
        this.setState({
          data: this.state.responseData[i].show,
        });
        localStorage.setItem('id-details', urlID2);
        localStorage.setItem('data-details', JSON.stringify(this.state.responseData[i].show));
      }
    }
  }


  render() {
    return (
      <div className='details' ref={div => this.detailsDOMElement = div}>
        <h3 className='details__title'>Details of TV-Show</h3>
        <p className='details__promo'>More detailed information about the select TV show</p>
        <div className="details__wrap">

          <div className="details__content">
            <img className='details__img'
                 src={this.state.data.image ? this.state.data.image.original : 'https://via.placeholder.com/420x650'} alt='img'/>
            <ul className='details__items'>
              <li className="details__item">
                <div className="genres">Genres:  &#8194;
                  {this.state.data.genres.length ? this.state.data.genres.join(' | ') : '-'}
                </div>
              </li>
              <li className="details__item">
                <div className="language">Language:  &#8194;
                  {this.state.data.language ? this.state.data.language : '-'}
                </div>
              </li>
              <li className="details__item">
                <div className="name">Name:  &#8194;
                  {this.state.data.name ? this.state.data.name : '-'}
                </div>
              </li>
              <li className="details__item">
                <div className="officalSite">Official Site:  &#8194;
                  <a href={this.state.data.officialSite} target="_blank">
                    {this.state.data.officialSite ? this.state.data.officialSite : ''}
                  </a>
                </div>
              </li>
              <li className="details__item">
                <div className="premiered">Premiered:  &#8194;
                  {this.state.data.premiered ? this.state.data.premiered : '-'}
                </div>
              </li>
              <li className="details__item">
                <div className="rating">Rating:  &#8194;
                  {this.state.data.rating.average ? this.state.data.rating.average : '-'}
                </div>
              </li>
              <li className="details__item">
                <div className="runtime">Runtime:  &#8194;
                  {this.state.data.runtime ? this.state.data.runtime : '-'} min
                </div>
              </li>
              <li className="details__item">
                <div className="schedule">Schedule:  &#8194;
                  <span className="days">
                    {this.state.data.schedule.days[0] ? this.state.data.schedule.days[0] : '-'}
                  </span>
                  <span className="Time">
                      &#8194;at  &#8194; ({this.state.data.schedule.time ? this.state.data.schedule.time : '-'})
                  </span>
                </div>
              </li>
              <li className="details__item">
                <div className="status">Status:  &#8194;
                  {this.state.data.status ? this.state.data.status : '-'}
                </div>
              </li>
              <li className="details__item">
                <div className="type">Type:  &#8194;
                  {this.state.data.type ? this.state.data.type : '-'}
                </div>
              </li>
            </ul>
          </div>

          <div className='details__description'>
            <div className="summary">Summary:  &#8194;
              {this.state.data.summary ? this.state.data.summary.replace(/(<\/?\w+>){1,}/gm, " ") : '-'}
            </div>
          </div>

        </div>
      </div>
    );
  }
};
