// pages/Login/index.js
import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import Input from '../../components/Input/index';
import Button from '../../components/Button/index';
import './index.scss';

export default class extends Component {
  onSubmit = (event) => {
    event.preventDefault();
    const userName = this.usernameInput.value;
    const password = this.passwordInput.value;
    const localUser = JSON.parse(localStorage.getItem('user'));

    this.props.onLogin({userName, password});

    if (!localUser) return;

    if (userName === localUser.userName && password !== localUser.password) {
      this.form.classList.add('login-error');
      this.popupError.classList.add('popup-error');
      setTimeout(() => {
        if (!this.popupError) return;
        this.popupError.classList.remove('popup-error');
      }, 2500);
    }
  };

  render() {
    return (
      <div className="wrapper-login">
        <form onSubmit={this.onSubmit} className="login-regist__form login" ref={(form) => this.form = form}>
          <div className="popup" ref={(div) => this.popupError = div}>sorry incorrect password</div>
          <Input
            type="username"
            name="username"
            placeholder="Username"
            required
            refs={(input) => this.usernameInput = input}
          />
          <Input
            type="password"
            name="password"
            placeholder="Password"
            required
            refs={(input) => this.passwordInput = input}
          />

          <Link className="login-regist__link" to="/registration">Don't have an account?</Link>
          <Button text="Log in" />
        </form>
      </div>
    );
  }
}
