// pages/Registration/index.js
import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import Input from 'components/Input/index';
import Button from 'components/Button/index';
import './index.scss';
import myHistory from 'myHistory';

export default class extends Component {
  userRegister = (obj) => {
    const strObj = JSON.stringify(obj);
    localStorage.setItem('user', strObj);
  };

  onSubmit = (event) => {
    event.preventDefault();
    this.userRegister({
      firstName: this.firstName.value,
      lastName: this.lastName.value,
      email: this.email.value,
      userName: this.userName.value,
      password: this.password.value,
      birthday: this.birthday.value,
      phone: this.phone.value,
      radioMale: this.radioMale.checked,
      radioFemale: this.radioFemale.checked,
      select: this.select.value
    });

    localStorage.setItem('login', '1');
    return myHistory.push('/');
  };

  render() {
    return (
      <div className="wrapper-reigst">
        <form onSubmit={this.onSubmit} className="login-regist__form regist">
          <Input
            type="text"
            name="firstname"
            placeholder="First name"
            required
            refs={(input) => this.firstName = input}
          />
          <Input
            type="text"
            name="lastname"
            placeholder="Last name"
            required
            refs={(input) => this.lastName = input}
          />
          <Input
            type="email"
            name="email"
            placeholder="Email"
            required
            refs={(input) => this.email = input}
            pattern="\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,})+"
          />
          <Input
            type="username"
            name="username"
            placeholder="Username"
            // required={true}
            refs={(input) => this.userName = input}
          />
          <Input
            type="password"
            name="password"
            placeholder="Password [0-9a-zA-Z!@#$%^&*]"
            required
            pattern="(?=.*[0-9])(?=.*[!@#$%^&*]?)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z!@#$%^&*]{6,}"
            refs={(input) => this.password = input}
          />
          <Input
            type="text"
            name="birthday"
            placeholder="YYYY/MM/DD"
            required
            pattern="[0-9]{4}\/[0-9]{2}\/[0-9]{1,2}"
            refs={(input) => this.birthday = input}
          />
          <Input
            type="phone"
            name="phone"
            placeholder="+38-XXX-XXX-XX-XX"
            required
            refs={(input) => this.phone = input}
            pattern="^(\s*)?(\+)?([- _():=+]?\d[- _():=+]?){10,14}(\s*)?$"
          />

          <div className="wrap-other-input">
            <div className="wrap-other-input-left">
              <label>
                Male
                <Input
                  type="radio"
                  name="gender"
                  value="male"
                  check
                  required
                  refs={(input) => this.radioMale = input}
                />
                <span className="slider" />
              </label>

              <label>
                Female
                <Input
                  type="radio"
                  name="gender"
                  value="female"
                  required
                  refs={(input) => this.radioFemale = input}
                />
                <span className="slider" />
              </label>
            </div>

            <div className="wrap-other-input-right">
              <label>
                Family status:
                <select name="family-status" ref={(input) => this.select = input}>
                  <option value="married">Married</option>
                  <option value="unmarried">Unmarried</option>
                  <option value="active">Actively looking</option>
                  <option value="developer">Developer</option>
                </select>
              </label>
            </div>
          </div>

          <Link to="/login" className="login-regist__link">
            Back to login
          </Link>

          <Button text="Sign up" />
        </form>
      </div>
    );
  }
}
