// components/FormRegistration/index.js
import React from "react";
import {Link} from "react-router-dom";
import Input from "components/Input";
import Button from "components/Button";
import './index.scss';

export default (props) => {
  return (
    <div className="wrapper-reigst">
      <form onSubmit={props.onSubmit} className="login-regist__form regist">
        <label>
          <Input
            type="text"
            name="firstname"
            placeholder="First name"
            // required
            onChange={e => props.onFirstName(e.target.value)}
            // refs={(input) => this.firstName = input}
          />
          <span className='text-error'>{props.onErrorFirstName}</span>
        </label>
        <label>
          <Input
            type="text"
            name="lastname"
            placeholder="Last name"
            // required
            // refs={(input) => this.lastName = input}
            onChange={e => props.onLastName(e.target.value)}
          />
          <span className='text-error'>{props.onErrorLastName}</span>
        </label>
        <label>
          <Input
            type="email"
            name="email"
            placeholder="Email"
            onChange={e => props.onEmail(e.target.value)}
            // required
            // refs={(input) => this.email = input}
            // pattern="\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,})+"
          />
          <span className='text-error'>{props.onErrorEmail}</span>
        </label>
        <label>
          <Input
            type="username"
            name="username"
            placeholder="Username"
            onChange={e => props.onUserName(e.target.value)}

            // required={true}
            // refs={(input) => this.userName = input}
          />
          <span className='text-error'>{props.onErrorUserName}</span>
        </label>
        <label>
          <Input
            type="password"
            name="password"
            placeholder="Password [0-9a-zA-Z!@#$%^&*]"
            onChange={e => props.onPassword(e.target.value)}
            // required
            // pattern="(?=.*[0-9])(?=.*[!@#$%^&*]?)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z!@#$%^&*]{6,}"
            // refs={(input) => this.password = input}
          />
          <span className='text-error'>{props.onErrorPassword}</span>
        </label>
        <label>
          <Input
            type="text"
            name="birthday"
            placeholder="MM/DD/YYYY"
            onChange={e => props.onBirthday(e.target.value)}
            // required
            // pattern="[0-9]{4}\/[0-9]{2}\/[0-9]{1,2}"
            // refs={(input) => this.birthday = input}
          />
          <span className='text-error'>{props.onErrorBirthday}</span>
        </label>
        <label>
          <Input
            type="phone"
            name="phone"
            placeholder="+38-XXX-XXX-XX-XX"
            onChange={e => props.onPhone(e.target.value)}
            // required
            // refs={(input) => this.phone = input}
            // pattern="^(\s*)?(\+)?([- _():=+]?\d[- _():=+]?){10,14}(\s*)?$"
          />
          <span className='text-error'>{props.onErrorPhone}</span>
        </label>


        <div className="wrap-other-input">
          <div className="wrap-other-input-left">
            <label>
              Male
              <Input
                type="radio"
                name="gender"
                value="male"
                onChange={e => props.onRadioMale(e.target.checked)}
                check
                // required
                // refs={(input) => this.radioMale = input}
              />
              <span className="slider" />
            <span className='text-error'>{props.onErrorRadioMale}</span>
            </label>

            <label>
              Female
              <Input
                type="radio"
                name="gender"
                value="female"
                onChange={e => props.onRadioFemale(e.target.value)}
                // required
                // refs={(input) => this.radioFemale = input}
              />
              <span className="slider" />
              <span className='text-error'>{props.onErrorRadioFemale}</span>
            </label>
          </div>

          <div className="wrap-other-input-right">
            <label>
              Family status:
              <select
                defaultValue='active' name="family-status">
                <option value="married">Married</option>
                <option value="unmarried">Unmarried</option>
                <option value="active">Actively looking</option>
                <option value="developer">Developer</option>
              </select>
              <span className='text-error'>{props.onErrorSelect}</span>
            </label>
          </div>
        </div>

        <Link to="/login" className="login-regist__link">
          Back to login
        </Link>

        <Button text="Sign up" />
      </form>
    </div>
  );
}
